### PR TITLE
Add `lightning_app` docs

### PR DESCRIPTION
**Pull request by tchaton**
## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

**Review by rohitgr7 — APPROVED** [review]

File: .github/workflows/docs-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          persist-credentials: false
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0'
+        with:
+          credentials_json: ${{ secrets.GCS_SA_KEY }}
```
Comment (manskx) [inline .github/workflows/docs-deploy.yml:24]: do we have this configured?

File: .actions/app/delete_cloud_lightning_apps.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import os
```
Comment (manskx) [inline .actions/app/delete_cloud_lightning_apps.py:1]: This is not related to the docs, should be in another PR. I added it with the e2e tests 😄 

https://github.com/Lightning-AI/lightning/pull/13499/files#diff-25803814a480f40540e95e6dc62c8dd0995ad5eedfbf858ab5a28d52c27516b6R331

File: .actions/app/download_frontend.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import lightning_app
```
Comment (manskx) [inline .actions/app/download_frontend.py:1]: same ^^ unrelated to the docs

File: .github/workflows/docs-deploy.yml (hunk 1 above)
Comment (tchaton) [reply .github/workflows/docs-deploy.yml:24]: Yes, this was added by Lexie. But we can try this only once merged.

File: .actions/app/delete_cloud_lightning_apps.py (hunk 2 above)
Comment (tchaton) [reply .actions/app/delete_cloud_lightning_apps.py:1]: Removed.

File: .actions/app/download_frontend.py (hunk 3 above)
Comment (tchaton) [reply .actions/app/download_frontend.py:1]: Removed !

**Review by justusschock — APPROVED** [review]

**Review by manskx — APPROVED** [review]

File: src/lightning_app/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+## \[0.5.2\] - 2022-MM-DD
+
+### Added
+
+- Update the Lightning App docs ([#13537](https://github.com/PyTorchLightning/pytorch-lightning/pull/13537))
```
Comment (carmocca) [inline src/lightning_app/CHANGELOG.md:11]: This wouldn't be a CHANGELOG addition

File: src/lightning_app/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+## \[0.5.2\] - 2022-MM-DD
```
Comment (carmocca) [inline src/lightning_app/CHANGELOG.md:7]: I don't think we have those in the other CHANGELOG

```suggestion
## [0.5.2] - 2022-MM-DD
```

And isn't next release 0.6?

**Review by carmocca — APPROVED** [review]

File: src/lightning_app/CHANGELOG.md (hunk 4 above)
Comment (tchaton) [reply src/lightning_app/CHANGELOG.md:11]: This is almost the entire documentation. I believed it can be added as we have an empty changelog and the PR has 7k lines.

File: src/lightning_app/CHANGELOG.md (hunk 5 above)
Comment (tchaton) [reply src/lightning_app/CHANGELOG.md:7]: We could do both. This hasn't been decided. I would prefer weekly releases.

**Review by awaelchli — APPROVED** [review]
